### DISCUSSION TITLE
Not entirely sure how I feel about the redesign

### DISCUSSION THREAD

**Discussion by Der-Flip** [Yet-Zio/yetCalc#155] [General]
First off: I know and realize I am not good with change. I prefer things to stay as they are, especially if they have been pretty good already.

I also wanna say that I understand and appreciate things move ahead, designs change and get updated to the current popular design language.

So please don't see my criticism as nagging, entitled begging to return to the old, an attack or anything like that. It's just me wanting to give my two cents.

First of all: I think it is great to have the scientific tools available in portrait mode now. I have my phone locked to portrait 98% of the time and this makes it so much easier.

Some things that bother me though:

the icon.
I understand one needs a bit of an icon the stands out and is unique. Sadly though, the now new icon doesn't really communicate calculator for me. I also think the contrasts are too low making it a dark blob with details that are difficult to spot at quick glance. I think bigger operator symbols and better colour contrast would improve the design already.

The general calculator UI:
Whether there should be buttons visible or not is something that is so dependent on taste I won't mark it as a flaw, really. Personally I like some way to signify where the button ends but again, that's personal preference.
On the bottom though there is a huge amount of wasted space. There could easily fit another row of buttons or one could decompress the top a bit.
Depending on what's valued more I would align the number pad to the bottom, gaining an entire row up top. I would move the AC button into that gained top row, as well as make the arrow for the scientific tools a button on the right side top row, opposite the AC.
The gained free space I would use to reintroduce the square root.
I know, this layout remains unchanged from the previous version but the previous version did include a square root and was a lot more decompressed up top giving the (temp) results and operation much more breathing room.

I like the distinguishing between keypad and parentheses and history buttons better in the new version, and while I do agree it does follow a certain logic to put the backspace up there too, I like it better down near the keypad so I can correct without having to reach all the way up there, which is especially difficult if you only have one hand available that moment.
I'm further not entirely sure why you have (.) and (,). I would, in the settings create an entry what the decimal should look like then automatically insert the thousands indicator when typing / displaying results. For consistency with other numpads I would further move the 0 beneath the 2, rather than the one. With that change though, I agree with you, I wouldn't know where else to put the backspace as the decimal should remain right of the 0 and placing the backspace into that opened up space left of the zero wouldn't be nice either. So I'd probably keep the layout as is, exchanging one of the ,/. For the backspace. But I also understand having the backspace next to the = isn't great practice either \^\^
(Why is the 0 bold, btw?)

Overall it's not a bad design, it is actually pretty sleek looking, I'm just not great with change and a few points seem improvable. Because of my (probably unhealthy) stubbornness I will remain on 1.1.4 for now.

Hope that this wall of text isn't too inconsiderate and rude. This really isn't my intention.

**Comment by Yet-Zio** [comment]
>So please don't see my criticism as nagging, entitled begging to return to the old, an attack or anything like that. It's just me wanting to give my two cents.

>First of all: I think it is great to have the scientific tools available in portrait mode now. I have my phone locked to portrait 98% of the time and this makes it so much easier.

Hi @Der-Flip, I think it's good to state your opinion on the changes, there's nothing wrong with that and I do appreciate what you as a user think about these changes. 
Yea, always wanted to bring the scientific tools to portrait mode ( Some users actually didn't know there was a scientific mode 😂), this makes up for it kinda :).

> the icon.
I understand one needs a bit of an icon the stands out and is unique. Sadly though, the now new icon doesn't really communicate calculator for me. I also think the contrasts are too low making it a dark blob with details that are difficult to spot at quick glance. I think bigger operator symbols and better colour contrast would improve the design already.

Yea, I know it looks kinda weird, not typical of a calculator to look like that. When you look at most of the calculator apps nowadays, the design they always have in common with the icon are primitive operations like +, -, *, /.

Just fyi, this might be long 😅. The old icon I designed had the same elements, with words in some corners. But then I thought about it and what more can a calculator represent and yetCalc in general. So, since yetCalc had all these modes, calculator, converter, programmer, ... I decided I want to make it simpler, with no words and such that, just an icon representing what yetCalc can do but still need the primitive operator symbols, etc.

So I put them in sides and in the middle I wanted to represent the app with something so that users know its 'yetCalc', so I put the 'equal to' along with the 'infinite' symbol and then the final touch was the "Y" from yetCalc. I found it funny how it looked tho, like a face but also representing what it means with infinite possibilities. Yea and about the contrast, I agree that the colors are quite dark and might not be sufficient on some devices, the new dark colors are from the app(the dark green), however if you switch to themed icons in your system settings, the icon might look a little bit better depending on the color and system theme.

> On the bottom though there is a huge amount of wasted space. There could easily fit another row of buttons or one could decompress the top a bit.

Yep, that space is really an issue on larger devices, even medium ones. But I didn't want to replicate the same old issues of buttons getting squashed, cut off, etc due to screen sizes, In web development or desktop, this is pretty easy to manage, in terms of mobile devices, this is kinda of a tough thing to do as our phones come in different sizes and it is really hard to manage all that with a page that requires all the buttons to show up, especially a calculator and since Im doing it in native kotlin with XML Layouts, not Jetpack Compose or Flutter, this is even harder as I might need to create layout and sizes for each and every device which is a hazzle. Therefore I did not purposely chain it to the bottom, so that even if its not same on all devices but atleast it can be useful. 

I can maybe in a future update chain it to the bottom programmatically depending on the screen size (hmm, writing this I realize I should have had that idea earlier, damn).

> I know, this layout remains unchanged from the previous version but the previous version did include a square root and was a lot more decompressed up top giving the (temp) results and operation much more breathing room.

>I like the distinguishing between keypad and parentheses and history buttons better in the new version, and while I do agree it does follow a certain logic to put the backspace up there too, I like it better down near the keypad so I can correct without having to reach all the way up there, which is especially difficult if you only have one hand available that moment.
I'm further not entirely sure why you have (.) and (,). I would, in the settings create an entry what the decimal should look like then automatically insert the thousands indicator when typing / displaying results.

Ah I see, the point separator is actually for providing decimal numbers directly and the other (,) comma is for certain functions, for example there's nPr(,), nCr(,), Stril1(), integral and diff which have multiple parameters and require comma for separation, so if you like accidentally remove a comma, you might need to input it back. But mostly, many users may not use those functions at all and the comma itself.

Yea, I agree that it was easier to access the square root button that way, but I needed to bring it back in a reusable way without hindering the other buttons. Perhaps a setting can be provided to replace one of the buttons in the UI, the comma maybe, with not only backspace, any common buttons you desire, I think that would be a good setting.

> (Why is the 0 bold, btw?)

I had that since the old design, 2 years back, I kept that to maybe highlight the zero. Forgot the exact reason why, but yea in this design, just wanted to keep the feel that its still the same old calc.

> Overall it's not a bad design, it is actually pretty sleek looking, I'm just not great with change and a few points seem improvable. 

Really thanks for your nice words, it helps a lot too.

> Because of my (probably unhealthy) stubbornness I will remain on 1.1.4 for now.

😁 I can totally understand that feel. Sometimes the older apps, the design and games just hit differently.

> Hope that this wall of text isn't too inconsiderate and rude. This really isn't my intention.

Again as I say, there's nothing wrong with that. As a user you can voice your opinion however you want and I do agree with most of your points and I will take it into consideration for the next set of changes for yetCalc.

Thank You

**Comment by neilcrownis** [comment]
On the subject of redesign. After the redesign, I got this bug. Or is it not a bug? At the very least, I would like to see the system keyboard appear when a key is pressed, not by itself. I just don't know how to describe it properly for an issue 😅😅😅

![Screenshot_20241210-203247_yetCalc.png](https://github.com/user-attachments/assets/bb1b51b4-c6cb-40ca-a112-3a3e7ee6686b)

**Comment by Yet-Zio** [comment]
Don't know how that could have happened even though I have disabled system keyboard on interaction. Did the keyboard come up when you selected part of the text or when you switched between another app?

**Reply by neilcrownis** [reply]
The keyboard appears if I switch from another application. It always appears. This does not happen when I open it normally. I can try to take logs and create an issue
